### DISCUSSION TITLE
I am unable to compile v1.26.0,how should I handle it

### DISCUSSION THREAD

**Discussion by ytjoe** [filecoin-project/lotus#11850] [General Q&A] [answered]
Intel(R) Xeon(R) CPU E5-2678 v3 @ 2.50GHz
cargo 1.77.1 (e52e36006 2024-03-26)
go version go1.21.7 linux/amd64

# Error log
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git.5f86ff9a6" -o lotus ./cmd/lotus
# runtime/internal/sys
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:9:7: len8tab redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:76:7: other declaration of len8tab
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:32:6: Len64 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:99:6: other declaration of Len64
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:50:7: m0 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:117:7: other declaration of m0
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:51:7: m1 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:118:7: other declaration of m1
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:52:7: m2 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:119:7: other declaration of m2
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:55:6: OnesCount64 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:122:6: other declaration of OnesCount64
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:86:6: LeadingZeros64 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:153:6: other declaration of LeadingZeros64
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:89:6: LeadingZeros8 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:156:6: other declaration of LeadingZeros8
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:92:6: Len8 redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:159:6: other declaration of Len8
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:101:6: Prefetch redeclared in this block
	/usr/local/go/src/runtime/internal/sys/intrinsics.go:200:6: other declaration of Prefetch
/usr/local/go/src/runtime/internal/sys/intrinsics_common.go:101:6: too many errors
make: *** [Makefile:89: lotus] Error 1

**Comment by rvagg** [comment, marked as answer]
@ytjoe perhaps try reinstalling Go, and before you do, clear out the old one. It looks to me like you have old and a new files from a Go upgrade. `rm -rf /usr/local/go` might help before you install again.

In fact, https://go.dev/doc/install even recommends deleting this directory if installing from a tarball download.

You could also try installing from an OS distro package instead to have this managed for you. I personally use https://snapcraft.io/go mostly to keep it updated and compatible.

**Reply by ytjoe** [reply]
Yes, that's the problem. I solved it through this method. Thank you
